calculator_test.go

```diff
@@ -2,23 +2,230 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
+func TestCloseEnough(t *testing.T) {
+	testCases := []struct {
+		a, b, roundedResult, tolerance float64
+		want                           bool
+	}{
+		{a: 5, b: 3, roundedResult: 1.6666, tolerance: 0.0001},
+		{a: 2, b: 3, roundedResult: 0.6666, tolerance: 0.001},
+		{a: 1, b: 3, roundedResult: 0.3333, tolerance: 0.01},
+	}
+
+	for _, tc := range testCases {
+
+		result, err := calculator.Divide(tc.a, tc.b)
+		if err != nil {
+			t.Fatalf("%v", err)
+		}
+
+		if !calculator.CloseEnough(tc.roundedResult, result, tc.tolerance) {
+			t.Errorf("Result(%g) is not close enough(%.5f)", tc.roundedResult, tc.tolerance)
+		}
+
+	}
+}
+
 func TestAdd(t *testing.T) {
```

Comment:
Suppose we combine the tests for Add, Subtract, and Multiply now—they're all very similar, aren't they?

Reply:
Implemented in [this](https://github.com/bitfieldconsulting/ftl-fundamentals-pmamberti/commit/249722f89f3e3da4fd45baca8b686ef52c3afd8c).

I have kept the separate tests too.